metis-pattern-analysis/src/main/java/eu/europeana/patternanalysis/ProblemPatternAnalyzer.java

```diff
@@ -53,51 +71,176 @@ public List<ProblemPattern> analyzeRecord(String rdfString) throws Serialization
+    constructProblemPattern(rdfAbout, ProblemPatternDescription.P9, checkP9(descriptions)).ifPresent(problemPatterns::add);
+    constructProblemPattern(rdfAbout, ProblemPatternDescription.P12, checkP12(titles)).ifPresent(problemPatterns::add);
     return problemPatterns;
   }
 
+  private static boolean isProviderProxy(ProxyType proxy) {
+    return proxy.getEuropeanaProxy() == null || isFalse(proxy.getEuropeanaProxy().isEuropeanaProxy());
+  }
+
   private List<ProxyType> getProviderProxies(RDF rdf) {
-    return rdf.getProxyList().stream().filter(proxyType -> proxyType.getEuropeanaProxy() != null)
-              .filter(not(proxyType -> proxyType.getEuropeanaProxy().isEuropeanaProxy())).collect(Collectors.toList());
+    return Optional.ofNullable(rdf.getProxyList()).stream().flatMap(Collection::stream)
+                   .filter(Objects::nonNull).filter(ProblemPatternAnalyzer::isProviderProxy)
+                   .collect(Collectors.toList());
   }
 
+  /**
+   * Check whether there is a title - description pair for which the values are equal, ignoring letter (upper or lower) case.
+   * <p>It will report a single occurrence for multiple same fields</p>
+   *
+   * @param titles the list of titles
+   * @param descriptions the list of descriptions
+   * @return the list of problem occurrences encountered
+   */
   private List<ProblemOccurrence> checkP2(List<String> titles, List<String> descriptions) {
-    final Set<String> uniqueTitles = titles.stream().map(String::toLowerCase).collect(Collectors.toSet());
-    final Set<String> uniqueDescriptions = descriptions.stream().map(String::toLowerCase).collect(Collectors.toSet());
+    final Set<String> uniqueTitles = titles.stream().map(String::toLowerCase).collect(toSet());
+    final Set<String> uniqueDescriptions = descriptions.stream().map(String::toLowerCase).collect(toSet());
     final HashSet<String> equalTitlesAndDescriptions = new HashSet<>(uniqueTitles);
     equalTitlesAndDescriptions.retainAll(uniqueDescriptions);
 
     return equalTitlesAndDescriptions.stream().map(
         value -> new ProblemOccurrence(format("Equal(lower cased) title and description: %s", value))
-    ).collect(Collectors.toList());
+    ).collect(toList());
   }
 
+  /**
+   * Check whether there is a title - description pair for which the values are too similar.
+   * <p>Blank values are filtered out. Titles and descriptions that are equal, ignoring letter (upper or lower) case are filtered
+   * out. Same titles will be reported once and will not have a duplicate of it self with same near identical descriptions</p>
+   *
+   * @param titles the list of titles
+   * @param descriptions the list of descriptions
+   * @return the list of problem occurrences encountered
+   */
+  private List<ProblemOccurrence> checkP3(List<String> titles, List<String> descriptions) {
+    final Map<String, List<String>> nearIdenticalTitleDescriptionsMap =
```

Comment:
:heavy_check_mark: let us see if 0.9 is a right accuracy measure, I saw in the tests in an excel sheet up to three decimals in the results.

Reply:
I've followed the documentation which specifically says to check for `>= 0.9` not looking at other decimal points after the first one.

Reply:
ok :+1: